### PR TITLE
Fix/19904 datepicker depend on admin settings

### PR DESCRIPTION
**Pull request by knight-bubble**
https://community.openproject.org/work_packages/19904

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://hirtie-maxim-fix-19904-datepicker-depend-on-admin-settings.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hirtie-maxim-fix-19904-datepicker-depend-on-admin-settings.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by floriank** [comment]
@hirtie-maxim please rebase

**Comment by floriank** [comment]
also, specs fail.

File: frontend/app/misc/datepicker-defaults.js [hunk 1]
```diff
@@ -64,8 +60,8 @@ jQuery(function($) {
     }
   };
 
-  if (CS.firstWeekDay && CS.firstWeekDay !== '') {
-    defaults.firstDay = parseInt(CS.firstWeekDay, 10);
+  if (regional && regional.firstDay) {
```
Comment (floriank) [inline frontend/app/misc/datepicker-defaults.js:63]: `regional.firstDay` can be `0`, therefore this can fail unexpectedly

**Comment by floriank** [comment]
Superseded by #2997

**Pull request closed without merging**